### ISSUE TITLE
Types error Namespace '"mongoose"' has no exported member 'DocumentDefinition'

### ISSUE THREAD

**Issue by kpachbiu88** [typegoose/typegoose#623]
![image](https://user-images.githubusercontent.com/2962737/137547698-0f4e2909-0476-4e1c-b14d-749595112073.png)

## What is the Problem?

Create repository with mongoose (typegoose) models.
But have error types

```
node_modules/@typegoose/typegoose/lib/typegoose.d.ts:132:45 - error TS2694: Namespace '"mongoose"' has no exported member 'DocumentDefinition'.

132     raw: mongoose.SchemaDefinition<mongoose.DocumentDefinition<SchemaDefinitionType>>;

node_modules/@typegoose/typegoose/lib/typegoose.d.ts:139:57 - error TS2694: Namespace '"mongoose"' has no exported member 'DocumentDefinition'.

139     constructor(raw: mongoose.SchemaDefinition<mongoose.DocumentDefinition<SchemaDefinitionType>>, direct?: boolean);
```

```json
{
  "name": "mongoose-schemas",
  "version": "1.0.0",
  "description": "Mongoose schemas",
  "main": "./index.js",
  "types": "./index.d.ts",
  "scripts": {
    "build": "tsc",
    "preinstall": "npm run build",
    "typecheck": "tsc --noEmit"
  },
  "devDependencies": {
    "typescript": "^4.4.4"
  },
  "dependencies": {
    "@typegoose/auto-increment": "^1.0.0",
    "@typegoose/typegoose": "^9.1.0",
    "mongoose": "^6.0.11"
  }
}
```

**Comment by hasezoey** [comment]
Note: i edited you issue to have the package.json as a codeblock

---

TL;DR: mongoose did a breaking type change, stick to version `6.0.10` for now

i can reproduce this issue with mongoose `6.0.11`, but not `6.0.10` (or `6.0.9`), which means there was a breaking change in terms of types in `6.0.11`, which will need to be fixed with typegoose

i guess the change that was problematic was:

> perf(index.d.ts): remove some unnecessary definitions, streamline some union types to reduce number of instantiations #10349

Reproduction:
- create clean new package install (npm / yarn)
- install all required dependencies
- install mongoose `6.0.11`
- have at least one importing `@typegoose/typegoose`
- observe error

This will probably be fixed with typegoose 9.2.0, if its not breaking other types

---

PS: next time, please use the proper issue template format

**Comment by AishwaryaNangare** [comment]
adding DocumentSetOptions without generics instead of DocumentOptions works for me try this
